### DISCUSSION TITLE
Ability to render the diagram as a standalone html page

### DISCUSSION THREAD

**Discussion by ThePlenkov** [likec4/likec4#303] [Ideas]
Hi! export feature is amazing and we already enjoy it - but it builds static images, while the original idea of likec4 is rendering an interactive component with a semantic zoom.

Therefore I'd like to come with the following proposal:

1. We need to have a command which is going to build a site. Ideally separate html page per view.

what about `likec4 export --as html` or something like `likec4 build` building a website.

2. After that we'll be able to serve that site statically for example using Github/Gitlab pages

Once the diagram is available as a standalone html page, it can be used as iframe in tools like Confluence ( or any other )

For us having an enterprise confluence data server having cloud extensions is not possible and plugin set is very limited. So having ability to insert reference to iframe would be already a great thing if that page can render text-based arcihtecture diagram with a semantic zoom like this tool does.

**Comment by ThePlenkov** [comment]
Basically what I want to say. To export an image we already render some html page to be able to export a page snapshot via headless browser. Why not just to try to render such an html page statically too with a dedicated instruction?

**Comment by davydkov** [comment]
@ThePlenkov, great idea! I did not think about "bundle all together as single html" as an export, even though I already have it for [development purposes](https://github.com/likec4/likec4/tree/main/packages/diagrams/src-dev) (`yarn dev` in __packages/diagrams__ starts vite with rendered diagrams)

We may take the same approach "vite preset" -> codegen -> "vite build"

**Reply by ThePlenkov** [reply]
@davydkov correct. So what I see indeed dist contains compiled js files, but from what I see  - index.html is not there, it's kept in the source folder.

Unfortunately being backend dev I don't have that much experience with vite - but in my personal vitepress project when I build docs, it also builds me html files with all the assets - that's how I can serve diagrams static ( via Gitlab pages in our case ).
